UnitySDK/Assets/ML-Agents/Scripts/RewardProvider/IRewardProvider.cs

```diff
@@ -0,0 +1,36 @@
+namespace MLAgents.RewardProvider
+{
+    /// <summary>
+    /// Reward providers allow users to provide rewards for Agent behavior during training in order to
+    /// give hints on what types of actions are "better" than others based on an Agent's previous observation.
+    /// </summary>
+    public interface IRewardProvider
+    {
+        /// <summary>
+        /// Get an incremental reward to pass along to a trainer.  
+        /// </summary>
+        /// <returns></returns>
+        float GetIncrementalReward();
+
+        /// <summary>
+        /// Adds a scalar value to the current reward for this step.
+        /// </summary>
+        void AddReward(float reward);
```

Comment:
I am not sure the interface of IRewardProvider should have a Add/SetReward. The implementations of this interface could have their own ways of calculating the reward.

Reply:
I thought of this as well.  We can keep the `AddReward` API on Agent, but if it's not a (what is currently named) LowLevelReward provider, do we assert?